### PR TITLE
Refactor AppBar and Extract Index.tsx into Integration.tsx

### PR DESCRIPTION
**Pull request by AdrianAndersen**
With this PR, I do the following:
- Generalize AppBar to only contain display logic
- Extract and gather user config logic into a separate /configuration module
- Extract Index.tsx logic to Integration.tsx, to enable multiple Integrations in the future.

I strongly advise to look at this commit by commit, as the last commit moves the _entire_ index.tsx logic into a separate file.

- refactor(AppBar): extract Logo component
- refactor(configuration): introduce /configuration component directory
- refactor(ConfigBar): extract ConfigBar to separate component
- refactor(AppBar): generalize to only contain display logic
- refactor(AppBar): move to /utils
- refactor(MobileConfigUpdateBar): move into ConfigBar
- refactor(index): move config logic into ConfigBar
- refactor(index): extract Index.tsx logic to separate integration component

### CODE REVIEW

File: components/configuration/ConfigBar.tsx [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+                            <LoginIcon />
+                        </IconButton>
+                    </Tooltip>
+                </Box>
+            )}
+            <MobileConfigUpdateBar
+                visible={selectionChanged}
+                isLoadingConfig={isLoadingConfig}
+                onUpdateConfig={onUpdateConfig}
+                onUndoSelectionChanges={() => onUndoSelectionChanges}
```
Comment (mathiazom) [inline components/configuration/ConfigBar.tsx:223]:
```suggestion
                onUndoSelectionChanges={() => onUndoSelectionChanges()}
```

Comment (mathiazom) [inline components/configuration/ConfigBar.tsx:223]:
```suggestion
                onUndoSelectionChanges={onUndoSelectionChanges}
```

**Review by mathiazom — CHANGES_REQUESTED** [review]
Looks good, just the small fix and a rebase please :hugs:

File: components/configuration/ConfigBar.tsx (hunk 1 above)
Comment (AdrianAndersen) [reply components/configuration/ConfigBar.tsx]: Fixed 💮

**Comment by AdrianAndersen** [comment]
Dunzo!

**Review by mathiazom — APPROVED** [review]

**Pull request merged**